app/controllers/api/v1/books_controller.rb

```diff
@@ -0,0 +1,30 @@
+module Api
+  module V1
+    class BooksController < ApplicationController
+      before_action :set_book, only: [:show]
+
+      def index
+        @books = Book.all
+        render_paginated @books, each_serializer: BookSerializer
+      end
+
+      def show
+        @book = set_book
```

Comment:
```
@book = Book.find(params[:id])
return not_found if @book.nil?
render json: @book
```

and define in application controller
```
def not_found
  render json: { error: 'record not found' }, status: :not_found
end
```
or something alike
(you could use i18n for the error message but i consider it's not necessary)